SimG4CMS/Calo/src/CaloHitID.cc

```diff
@@ -9,6 +9,7 @@
 CaloHitID::CaloHitID(uint32_t unitID, double timeSlice, int trackID, uint16_t depth, float tSlice, bool ignoreTkID)
     : timeSliceUnit(tSlice), ignoreTrackID(ignoreTkID) {
   setID(unitID, timeSlice, trackID, depth);
+  theFineTrackID = -1;
```

Comment:
this could be done in the constructor initialization list

Reply:
Both fixed